### ISSUE TITLE
gyp ERR! stack Error: certificate not trusted

### ISSUE THREAD

**Issue by n0mer** [libxmljs/libxmljs#333]
Hi, after migration to 0.14.3 got this error:

```
Step 17 : RUN npm install --save wolfram && npm install
 ---> Running in 322fb5fb35e4

> libxmljs@0.14.3 install /hubot/node_modules/wolfram/node_modules/libxmljs
> node-gyp rebuild

gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: certificate not trusted
gyp ERR! stack     at Error (native)
gyp ERR! stack     at TLSSocket.<anonymous> (_tls_wrap.js:929:36)
gyp ERR! stack     at TLSSocket.emit (events.js:104:17)
gyp ERR! stack     at TLSSocket._finishInit (_tls_wrap.js:460:8)
gyp ERR! System Linux 3.13.0-57-generic
gyp ERR! command "node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /hubot/node_modules/wolfram/node_modules/libxmljs
gyp ERR! node -v v0.12.7
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok
npm ERR! Linux 3.13.0-57-generic
npm ERR! argv "node" "/usr/bin/npm" "install" "--save" "wolfram"
npm ERR! node v0.12.7
npm ERR! npm  v2.14.1
npm ERR! code ELIFECYCLE

npm ERR! libxmljs@0.14.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the libxmljs@0.14.3 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the libxmljs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls libxmljs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /hubot/npm-debug.log
The command '/bin/sh -c npm install --save wolfram && npm install' returned a non-zero code: 1
```

**Comment by n0mer** [comment]
maybe Amazon S3 SSL certs policy is the root cause , https://github.com/nodejs/node/issues/402

**Comment by n0mer** [comment]
this can be a cause (commit happened after 0.14.2, included in 0.14.3) : https://github.com/polotek/libxmljs/commit/0e7416ab0230617c4418a1d90d425925e02ecc0e

> Get Travis to build for iojs as well
The problem with node-gyp not working with iojs can be solved by using the
version shipped with iojs, not the version from the latest npm we install
separately.  See nodejs/node#433 for details.
So we update npm to latest upstream only for old versions of node.js,
not for the current version or for iojs.

**Comment by n0mer** [comment]
w/out npm updating itself `libxmljs` is building correctly. but with npm 2.14.1 it is not

```
Step 10 : RUN npm install -g npm
 ---> Running in dac514a030e3
/usr/bin/npm -> /usr/lib/node_modules/npm/bin/npm-cli.js
npm@2.14.1 /usr/lib/node_modules/npm
```
